OneSignalSDK/onesignal/src/main/java/com/onesignal/GenerateNotification.java

```diff
@@ -69,34 +53,39 @@
 
 import com.onesignal.OneSignalDbContract.NotificationTable;
 
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.lang.reflect.Field;
+import java.math.BigInteger;
+import java.net.URL;
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Random;
+
 import static com.onesignal.OSUtils.getResourceString;
 
 class GenerateNotification {
-   private static Context currentContext = null;
+
+   private static final String NOTIFICATION_ID = "notificationId";
+   private static final String DISMISSED = "dismissed";
+   private static final Class<?> NOTIFICATION_OPENED_ACTIVITY_CLASS = NotificationOpenedActivity.class;
+
    private static String packageName = null;
    private static Resources contextResources = null;
-   private static Class<?> notificationOpenedClass;
-   private static boolean openerIsBroadcast;
 
    private static class OneSignalNotificationBuilder {
       NotificationCompat.Builder compatBuilder;
       boolean hasLargeIcon;
    }
 
    private static void setStatics(Context inContext) {
-      currentContext = inContext;
-      packageName = currentContext.getPackageName();
-      contextResources = currentContext.getResources();
-
-      PackageManager packageManager = currentContext.getPackageManager();
-      Intent intent = new Intent(currentContext, NotificationOpenedReceiver.class);
-      intent.setPackage(currentContext.getPackageName());
-      if (packageManager.queryBroadcastReceivers(intent, 0).size() > 0) {
```

Comment:
Since this is a breaking change let's rebase this PR off the 4.0.0 branch.

Reply:
I rebased it